src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp

```diff
@@ -72,6 +61,101 @@ bool SmokeCoAlarmServer::SetExpressedState(EndpointId endpointId, ExpressedState
     return success;
 }
 
+bool SmokeCoAlarmServer::AutoSetExpressedState(EndpointId endpointId,
+                                               std::array<ExpressedStateEnum, kPriorityOrderLength> priorityOrder)
+{
+    ExpressedStateEnum currentExpressedState = ExpressedStateEnum::kNormal;
+
+    for (ExpressedStateEnum priority : priorityOrder)
+    {
+        switch (priority)
+        {
+        case ExpressedStateEnum::kSmokeAlarm: {
+            AlarmStateEnum currentSmokeAlarm;
+            VerifyOrReturnValue(GetSmokeState(endpointId, currentSmokeAlarm), false);
+            if (currentSmokeAlarm != AlarmStateEnum::kNormal)
+            {
+                currentExpressedState = ExpressedStateEnum::kSmokeAlarm;
```

Comment:
all these are the same as `priority` right? In which case we can replace things with `currentExpressedState == priority` throughout and maybe simplify code?